src/avt/Queries/Queries/avtXRayImageQuery.C

```diff
@@ -33,9 +33,38 @@
 #include <string>
 #include <vector>
 
-
 int avtXRayImageQuery::iFileFamily = 0;
 
+const int NUMFAMILYFILES = 9999;
+
+// 
+// Filename Scheme information and handling
+// 
+
+// To add new filename schemes, these are the changes to make:
+//    1) increment `NUM_FILENAME_SCHEMES` by however many filename schemes you are planning to add
+//    2) add new entries to the `filename_schemes` array
+//    3) add new constants for your filename schemes (make sure in the same order as in `filename_schemes`)
+//    4) add new cases where necessary (probably just in `avtXRayImageQuery::Execute` baseName setup)
+//    5) add them to `src/gui/QvisXRayImageQueryWidget.C` in the constructor.
+
+// a constant for how many valid filename schemes there are
+const int NUM_FILENAME_SCHEMES = 3;
+
+// member filenameScheme indexes this array.
+const char *filename_schemes[NUM_FILENAME_SCHEMES] = {"none", "family", "cycle"};
+
+// constants for each of the filename schemes
+const int NONE = 0;
```

Comment:
I feel like I discussed this with @cyrush before and we thought that making this an enum was not the right idea... but I can't remember the reason why. I could be wrong. I'm thinking of the output type stuff, which works fundamentally the same way.

Reply:
At a min, maybe recalling that reasoning and add here. IMHO, I can't see a down side to enum'ing it even if its also to be supported in yaml.

Reply:
I'm not a C++ enum expert, so I could use some guidance. I use the `filenameSchemeValid` function when parsing the yaml to make sure that users didn't send us garbage. Is there a clean way to do this with enums? I don't want to use 

```
if (ftype >= NONE && ftype <= CYCLE)
``` 

just in case the enum changes in the future.

Reply:
@markcmiller86 I think if we do go the enum route, we should do the same thing for the output type options, which may require more thinking. How about I leave as is for now and make a note in my x ray query continued iteration ticket to look at that in the future?